src/vm/nativelibrary.cpp

```diff
@@ -0,0 +1,42 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "common.h"
+#include "nativelibrary.hpp"
+#include "dllimport.h"
+
+#include <shlwapi.h>
+#include <stdlib.h>
+#include "dllimport.h"
+#include "field.h"
+#include "assemblyname.hpp"
+#include "eeconfig.h"
+#include "strongname.h"
+#include "interoputil.h"
+#include "frames.h"
+#include "typeparse.h"
+#include "stackprobe.h"
+
+
+//// static
+INT_PTR QCALLTYPE NativeLibrary::LoadLibrary(QCall::AssemblyHandle callingAssembly, LPCWSTR libraryName, BOOL searchAssemblyDirectory, DWORD dllImportSearchPathFlag)
+{
+    QCALL_CONTRACT;
+
+    HMODULE moduleHandle = nullptr;
+
+    BEGIN_QCALL;
+
+    Assembly* pAssembly = nullptr;
+    if (static_cast<DomainAssembly*>(callingAssembly) != nullptr)
```

Comment:
Uh this cast is very weird. Are you looking for the `dynamic_cast`? That is so slow and honestly should rarely be used. So I guess the question is, what are you trying to do? 😄